### PR TITLE
Add details on rebasing and merging

### PR DESCRIPTION
**Pull request by WilliamBZA**
As discussed, I've added a quick description of how PRs are to be merged and our commit standards.

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -22,3 +22,5 @@ The maintainers [watch](https://github.com/Particular/ServiceInsight/watchers) t
 - After a pull request is merged, the milestone representing the next version should be added to:
   - The issue(s) which the PR is addressing, *or*
   - The PR itself, if it is not addressing any pre-existing issue.
+- If merging using the Github UI, the pull requests must not be merged using the "Squash and Merge" option as the Squash and Merge option creates a ff mmerge.
```
Comment (adamralph) [inline MAINTAINERS.md]: We've disabled the alternative merge style options for this repo so we see this:

image

Do we still need to include this point?

File: MAINTAINERS.md [hunk 2]
```diff
@@ -22,3 +22,5 @@ The maintainers [watch](https://github.com/Particular/ServiceInsight/watchers) t
...
+- Each individual commit within a pull request must relate to a complete unit of work. If any "fix up" commits are made, they must be squashed into the pull request before merging.
```
Comment (adamralph) [inline MAINTAINERS.md]: The fixup commits aren't really squashed "into the pull request". I would simplify it to:

> they must be squashed before merging.

We may also want to call out that the workflow should like this:

- Reviewer: Request changes or comment.
- Submitter: Push fixup.
- (Repeat)
- Reviewer: Request squash.
- Submitter: Squash and force push.
- Reviewer: Approve.
- (Any maintainer): Merge.

File: MAINTAINERS.md (hunk 1 above)
Comment (HEskandari) [inline MAINTAINERS.md]: should be:

>...ff merge.

Comment (WilliamBZA) [reply MAINTAINERS.md]: I think it's worth including as an explanation of the reason why it was disabled. I'll change the text to match an explaining tone.

**Comment by WilliamBZA** [comment]
Updated, and added a summary section to show the workflow

File: MAINTAINERS.md (hunk 1 above)
Comment (adamralph) [reply MAINTAINERS.md]: > I'll change the text to match an explaining tone.

@WilliamBZA is that change still pending?

Comment (WilliamBZA) [reply MAINTAINERS.md]: Nope, I forgot to do it!

**Comment by WilliamBZA** [comment]
Squashed

**Pull request merged**